scrapy_statsd_extension/handlers.py

```diff
@@ -5,15 +5,56 @@
 
 class StatsdBase(object):
 
-    def __init__(self, host, port):
-        self.client = statsd.StatsClient(host, port)
+    def __init__(self, crawler_settings):
+        host = crawler_settings.get('STATSD_HOST', defaults.STATSD_HOST)
+        port = crawler_settings.get('STATSD_PORT', defaults.STATSD_PORT)
+        prefix = crawler_settings.get('STATSD_PREFIX', defaults.STATSD_PREFIX)
+        self.client = statsd.StatsClient(host, port, prefix)
+
+        self.prefixes_to_log = crawler_settings.get(
+            'STATSD_LOG_ONLY', defaults.STATSD_LOG_ONLY)
+        self.dont_log_all_fields = bool(log_only) 
+
+        self.ignored_prefixes = (
+            crawler_settings.get('STATSD_IGNORE', defaults.STATSD_IGNORE) or 
+            []
+        )
+
+        self.tagging_enabled = crawler_settings.get('STATSD_TAGGING', defaults.STATSD_TAGGING)
+        self.tags = crawler_settings.get('STATSD_TAGS', defaults.STATSD_TAGS)
 
     @classmethod
     def from_crawler(cls, crawler):
-        host = crawler.settings.get('STATSD_HOST', defaults.STATSD_HOST)
-        port = crawler.settings.get('STATSD_PORT', defaults.STATSD_PORT)
+        return cls(crawler.settings)
+
+    def not_ignored_field(self, key):
+        for prefix in self.ignored_prefixes:
+            if key.startswith(prefix):
+                return False
+
+        return True
+
+    def has_valid_prefix(self, key):
+        if self.dont_log_all_fields is False:
```

Comment:
It might be easier to read if you reverse this bool, and make it `log_all_fields`.